astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -47,6 +47,13 @@ def _from_tree_base_transform_members(cls, model, node, ctx):
                 param_and_model_constraints[constraint] = node[constraint]
         model._initialize_constraints(param_and_model_constraints)
 
+        if "input_units_equivalencies" in node:
+            # this still writes eqs. for compound, but operates on each sub model
+            if not isinstance(model, core.CompoundModel):
+                eqs = node['input_units_equivalencies']
+                eqs = yamlutil.tagged_tree_to_custom_tree(eqs, ctx)
```

Comment:
Now that #10189 has been merged, it's safe to remove calls to `tagged_tree_to_custom_tree` and `custom_tree_to_tagged_tree`.